client/src/components/Checkbox.jsx

```diff
@@ -1,19 +1,29 @@
-import React from "react";
+import React, { useState } from "react";
 import "./Checkbox.css";
 
-function Checkbox({ label, checked, value, onChange }) {
-  const handleChange = () => onChange(!checked);
-
+function Checkbox({ label, checked, value, onChange, labelBefore }) {
+  const [isChecked, setChecked] = useState(checked ? checked : false);
```

Comment:
1. What does managing the state here do for us? I guess it allows us to use this checkbox input in an uncontrolled way more basic JS way (so maybe Anders suggested this). If so I am fine with that but otherwise I don't see the point. 

2. `checked ? checked : false` is bad logic. if true, true else false. In cases like this better to just do type coercion. so all truthy values evaluate to true and falsy values evaluate to false. `!!checked`

Reply:
1. previously it had been modifying its props, so i reformatted to this. you're suggesting state would be better managed in a parent component?
2. not quite sure what you mean here, please explain.

Reply:
1. I think state managed in the parent component makes more sense. Data flows down. So keeping track of it outside of this component is necessary either way. Unless we were to use this component in some sort of form. 

2. `checked ? checked : false === !!checked` b/c checked is a boolean (or undefined). Using ternaries to return booleans doesn't make sense. boolean ? true : false can just be boolean. But we do !! for type coercion.